src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -82,12 +82,16 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
 
         # generate the ganesha config
         def get_ganesha_conf() -> str:
-            context = dict(user=rados_user,
-                           nodeid=f'{daemon_spec.service_name}.{daemon_spec.rank}',
-                           pool=spec.pool,
-                           namespace=spec.namespace if spec.namespace else '',
-                           rgw_user=rgw_user,
-                           url=spec.rados_config_location())
+            context = {
+                "user": rados_user,
+                "nodeid": f'{daemon_spec.service_name}.{daemon_spec.rank}',
+                "pool": spec.pool,
+                "namespace": spec.namespace if spec.namespace else '',
+                "rgw_user": rgw_user,
+                "url": spec.rados_config_location(),
+                "port": daemon_spec.ports[0] if daemon_spec.ports else 2049,
```

Comment:
Add a note saying `2049` is the default port used by ganesha or define constant for it `DefaultGaneshaPort=2049`